Exception with LinkedIn provider and play2-memcached plugin

We've switched our application cache from the default EHCache implementation to a memcached-based one, using the [play2-memcached](https://github.com/mumoshu/play2-memcached) plugin.

When trying to link a LinkedIn identity to an existing account, we're encountering the following exception:

```
play.api.Application$$anon$1: Execution exception[[NotSerializableException: com.feth.play.module.pa.user.EmploymentsIdentity$EmploymentInfo]]
	at play.api.Application$class.handleError(Application.scala:293) ~[play_2.10.jar:2.2.5]
	at play.api.DefaultApplication.handleError(Application.scala:399) [play_2.10.jar:2.2.5]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$3.apply(PlayDefaultUpstreamHandler.scala:264) [play_2.10.jar:2.2.5]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$3.apply(PlayDefaultUpstreamHandler.scala:264) [play_2.10.jar:2.2.5]
	at scala.Option.map(Option.scala:145) [scala-library.jar:na]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:264) [play_2.10.jar:2.2.5]
Caused by: java.io.NotSerializableException: com.feth.play.module.pa.user.EmploymentsIdentity$EmploymentInfo
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1183) ~[na:1.7.0_91]
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347) ~[na:1.7.0_91]
	at java.util.ArrayList.writeObject(ArrayList.java:742) ~[na:1.7.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_91]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_91]
```

I believe this is caused by the serialization engine used by the *play2-memached* plugin. The plugin uses Java serialization to serialize objects, and thus relies on the `Serializable` interface.

I have forked Play Authenticate, and intend to change `com.feth.play.module.pa.user.EmploymentsIdentity.EmploymentInfo` to implement the `Serializable` interface.

I'm raising this issue to make you aware of it, and ask if there would be interest in pulling this change back into master? 

I believe this is related to #276 , as that user is also using the *play2-memcached* plugin to add memcached support.

Affected Play version: 2.2.5

--- a/code/app/com/feth/play/module/pa/user/EducationsIdentity.java
+++ b/code/app/com/feth/play/module/pa/user/EducationsIdentity.java
@@ -1,10 +1,11 @@
 package com.feth.play.module.pa.user;
 
+import java.io.Serializable;
 import java.util.Collection;
 
 public interface EducationsIdentity {
   
-  public static class EducationInfo {
+  public static class EducationInfo implements Serializable {
     protected String id;
     protected String schoolName;
     protected String degree;

--- a/code/app/com/feth/play/module/pa/user/EmploymentsIdentity.java
+++ b/code/app/com/feth/play/module/pa/user/EmploymentsIdentity.java
@@ -1,9 +1,10 @@
 package com.feth.play.module.pa.user;
 
+import java.io.Serializable;
 import java.util.Collection;
 
 public interface EmploymentsIdentity {
-  public static class EmploymentInfo {
+  public static class EmploymentInfo implements Serializable {
     protected String id;
     protected String title;
     protected String summary;

